src/hp/dns.rs

```diff
@@ -0,0 +1,19 @@
+use trust_dns_resolver::TokioAsyncResolver;
+
+use once_cell::sync::Lazy;
+pub static DNS_RESOLVER: Lazy<TokioAsyncResolver> = Lazy::new(|| {
+    TokioAsyncResolver::tokio_from_system_conf().expect("unable to create DNS resolver")
+});
```

Comment:
ooh, nice i think.

(though the blank lines are a bit wonky)

Reply:
this file is all dig! fixing the formatting tho